generated/nixnetsocket/nixnetsocket.proto

```diff
@@ -82,15 +84,30 @@ message VirtualInterface {
   repeated GatewayAddress gateway_addresses = 8;
 }
 
+message IPv6Addr {
+  bytes addr = 1;
+}
+
+message IPv4Addr {
+  uint32 addr = 1;
+}
+
+message Addr {
+  oneof addr {
+    IPv4Addr ipv4 = 1;
+    IPv6Addr ipv6 = 2;
+  }
+}
+
 message SockAddrIPv4 {
   uint32 port = 1;
-  uint32 addr = 2;
+  IPv4Addr addr = 2;
```

Comment:
I was originally "collapsing" the xnet addr structs (`nxin_addr` and `nxin6_addr`) into the containing `sockaddr` structs (which also have port/etc). But, with the `aton`/`pton` APIs, we now have methods that populate these directly. So I updated sockaddr to have the additional struct level as well.

I'm also considering changing the names of these types, maybe to be closer to the nxsocket ones. I.e., InAddr, In6Addr, SockAddrIn, etc. I don't love those names (and am a little unsure on the capitalization) but I'm not convinced my names are an improvement. And the C API is out best touchstone. (I'm planning to save this for a future PR, if any, but would welcome feedback.

Reply:
Yeah... I think changing the names to be closer to the C API is probably the safe bet. At worst, the same and at best the client is familiar with the naming schemes from XNET C API or socket programming in general.